### PR TITLE
remove `backend` concept from Options

### PR DESCRIPTION
**Pull request by j042**
- for compatibility, kept backend argument (renamed to backends).
- added option for specifying default
- added tests

### CODE REVIEW

**Review by clemens-fricke — APPROVED** [review]
Looks good to me.

All examples run.
All tests were successful.

Some adaptations are needed in splinepy, but that is fine for this PR.


Before we increase the version number, I would like to look further into adding support for 2D and 3D graphics in `ipynb` natively in gustaf. I hope I will get to that before Christmas. I already have a spike running for this. I will need to test more thoroughly to be sure it works, but I am hopeful.

**Review by clemens-fricke — DISMISSED** [review]
Now all examples run again...

File: gustaf/helpers/options.py [hunk 1]
```diff
@@ -168,7 +234,7 @@ class ShowOption:
     specific common routines. ShowOption and ShowManager in a sense.
     """
 
-    __slots__ = ("_helpee", "_options", "_backend")
```
[the remark this answers is not in the document]
Comment (j042) [reply gustaf/helpers/options.py:171]: tataratat/splinepy#311

[the remark this answers is not in the document]
Comment (clemens-fricke) [reply gustaf/helpers/options.py:171]: thanks

**Pull request merged**